opm/output/eclipse/RestartIO.cpp

```diff
@@ -249,8 +255,9 @@ RestartValue load( const std::string& filename,
 
         if (ecl_file_view_has_kw( file_view , key.c_str())) {
             const ecl_kw_type * ecl_kw = ecl_file_view_iget_named_kw( file_view , key.c_str() , 0 );
-            std::vector<double> data = double_vector( ecl_kw );
-            rst_value.extra[ key ] = std::move(data);
+            const double * data_ptr = ecl_kw_get_double_ptr( ecl_kw );
```

Comment:
> I could not find where the optional keywords are set to be written with double?

Well - that is not configurable; extra keywords are always written in double.

> And how are they separated from the non-optional ones at write time 

Well, at write time everyting which is passed to `EclipseIO::writeRestartFile( )` is written to disk. I can not see that it makes sense to tell the writer - at write-time - that in a subsequent invocation down the line reading of this keyword will be optional?!

> Should we not read both float and double interchangably in any case?

We could of course decide to do that - the reason I was so keen on reverting the `float -> double` conversion was that I felt it was a crutch which enabled an inelegant "wrong" solution to the problem.